mobile/src/people/navigator/tab-bar-top-custom.component.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { ReactElement } from 'react';
+import { Employee } from '../../reducers/organization/employee.model';
+import { Department } from '../../reducers/organization/department.model';
+import { TabBarTop, TabBarTopProps, TabScene } from 'react-navigation';
+import { connect } from 'react-redux';
+import { AppState } from '../../reducers/app.reducer';
+import { Platform } from 'react-native';
+import { StyledText } from '../../override/styled-text';
+import tabBarStyles from '../../tabbar/tab-bar-styles';
+
+//============================================================================
+type Optional<P> = P | null | undefined;
+
+//============================================================================
+interface TabBarTopCustomProps {
+    employee: Employee | null;
+    department: Department | null;
+}
+
+//============================================================================
+class TabBarTopCustomImpl extends React.Component<TabBarTopProps & TabBarTopCustomProps> {
+
+    //----------------------------------------------------------------------------
+    public render(): React.ReactNode {
+        const { employee, department } = this.props;
+
+        return (
+            <TabBarTop
+                {...this.props}
+                getLabel={(scene) => {
+                    return this.getLabel(scene, employee, department);
+                }}
+            />
+        );
+    }
+
+    //----------------------------------------------------------------------------
+    private getLabel = (scene: TabScene, employee: Optional<Employee>, department: Optional<Department>): React.ReactNode | string => {
+        switch (scene.index) {
+            case 0:
+                return this.styleTabBarLabel(department ? department.abbreviation : 'Department');
+            case 1:
+                return this.styleTabBarLabel(employee ? 'Room ' + employee.roomNumber : 'Room');
```

Comment:
please, do not concatenate it using '+', use string interpolation instead. What's JavaScript, you might start World War III by summing string and some other type.

Reply:
Done. fbd889b2d55e4d7dc7d1ffa3646500c656be863d